cpp/path_module/algorithm/path.hpp

```diff
@@ -44,64 +45,113 @@ struct LabelSets {
 };
 
 struct LabelBools {
-  bool blacklisted = false;
-  bool terminated = false;
-  bool end_node = false;
-  bool whitelisted = false;
+  bool blacklisted = false;  // no node in the path will be blacklisted
+  bool terminated = false;   // returned paths end with a termination node but don't continue to be expanded further,
+                             // takes precedence over end nodes
+  bool end_node = false;     // returned paths end with an end node but continue to be expanded further
+  bool whitelisted = false;  // all nodes in the path will be whitelisted (except end and termination nodes)
+  // end and termination nodes don't have to respect whitelists and blacklists
```

Comment:
Comments are usually written above variable
```suggestion
   // no node in the path will be blacklisted
  bool blacklisted = false;
  // returned paths end with a termination node but don't continue to be expanded further,
  // takes precedence over end nodes
  bool terminated = false;   
   // returned paths end with an end node but continue to be expanded further                         
  bool end_node = false; 
  // all nodes in the path will be whitelisted (except end and termination nodes)
   // end and termination nodes don't have to respect whitelists and blacklists   
  bool whitelisted = false;  
 
```